src/checks/cargo/cargo-test.test.ts

```diff
@@ -0,0 +1,143 @@
+import { cargoTestCheck } from './cargo-test'
+import { cargoTestFailedOutput } from './resources/cargo-test-text'
+
+describe('checks/cargo-tests', () => {
+  it('converts no failed tests', () => {
+    const results = cargoTestCheck({ data: [], sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
+
+    expect(results).toEqual({
+      name: 'cargo test',
+      status: 'completed',
+      completed_at: expect.any(String),
+      head_sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a',
+      conclusion: 'success',
+      output: {
+        title: 'Tests passed',
+        summary: 'Tests passed'
+      }
+    })
+  })
+
+  it('converts invalid input', () => {
+    const results = cargoTestCheck({ data: {} as any, sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
+    expect(results).toEqual({
+      name: 'cargo test',
+      status: 'completed',
+      completed_at: expect.any(String),
+      head_sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a',
+      conclusion: 'skipped',
+      output: {
+        title: 'Unexpected test output',
+        summary: ''
+      }
+    })
+  })
+
+  it('converts failed compilation and tests', () => {
+    const results = cargoTestCheck({ data: cargoTestFailedOutput, sha: '11963e3cb7ecbb9247f638cc0fb047173a62cf7a' })
+
+    expect(results).toEqual({
```

Comment:
Here I would just use the following to make the test case less verbose:

``` typescript 
expect(results).toMatchSnapshot({
   completed_at: expect.any(String)
})
```

Reply:
I actually haven't touched this, but I can change it :)

Reply:
Ahh, so it was me who was a bit too quick :)